src/js/program/media-controller.js

```diff
@@ -0,0 +1,216 @@
+import cancelable from 'utils/cancelable';
+import { resolved } from 'polyfills/promise';
+import { MediaModel } from 'controller/model';
+import { seconds } from 'utils/strings';
+
+import { MEDIA_PLAY_ATTEMPT, MEDIA_PLAY_ATTEMPT_FAILED, PLAYER_STATE,
+    STATE_PAUSED, STATE_BUFFERING } from 'events/events';
+
+export default class MediaController {
+    constructor(provider, model) {
+        this.attached = true;
+        this.mediaModel = new MediaModel();
+        this.model = model;
+        this.provider = provider;
+    }
+
+    init(item) {
+        this.reset();
+        const { model, provider, mediaModel } = this;
+        const position = item ? seconds(item.starttime) : 0;
+        const duration = item ? seconds(item.duration) : 0;
+        const mediaModelState = mediaModel.attributes;
+        mediaModel.srcReset();
+        mediaModelState.position = position;
+        mediaModelState.duration = duration;
+        model.setProvider(provider);
+        model.setMediaModel(mediaModel);
+        // Initialize the provider last so it's setting properties on the (newly) active media model
+        provider.init(item);
+    }
+
+    reset() {
+        this.mediaModel = new MediaModel();
+    }
+
+    play(item, playReason) {
+        const { model, mediaModel, provider } = this;
+
+        if (!playReason) {
+            playReason = model.get('playReason');
+        }
+
+        model.set('playRejected', false);
+        let playPromise = resolved;
+        if (mediaModel.get('setup')) {
+            playPromise = provider.play();
+        } else {
+            mediaModel.set('setup', true);
+            playPromise = loadAndPlay(item, provider, model);
+            if (!mediaModel.get('started')) {
+                playAttempt(playPromise, model, playReason, provider);
+            }
+        }
+        return playPromise;
+    }
+
+    stop() {
+        const { provider } = this;
+        provider.stop();
```

Comment:
For consistency: `this.provider.stop();`

Reply:
Addressed